### PR TITLE
Revert to original config directory precedence but with improved documentation

### PR DESCRIPTION
**Pull request by philipl**
This is the straight revert of the three relevant changes, and improved documentation as a separate change. It didn't make sense to do a partial revert as the docs have to change substantially to reflect the actual behaviour.

### CODE REVIEW

File: DOCS/man/mpv.rst [hunk 1]
```diff
@@ -1501,6 +1501,30 @@ For Windows-specifics, see `FILES ON WINDOWS`_ section.
...
+    The standard configuration directory. This can be overridden by environment
+    variables:
+
+    If ``$XDG_CONFIG_HOME`` is set, then the derived configuration directory
+    will be ``$XDG_CONFIG_HOME/mpv``.
+
+    If ``$HOME`` is set, then the derived configuration directory will be
+    ``$HOME/.config/mpv``.
+
+    If ``$MPV_HOME`` is set, then the derived configuration directory will be
```
Comment (sfan5) [inline DOCS/man/mpv.rst:1514]: Isn't MPV_HOME preferred over all others? Is this list even ordered by precedence?

File: DOCS/man/mpv.rst [hunk 2]
```diff
@@ -1501,6 +1501,30 @@ For Windows-specifics, see `FILES ON WINDOWS`_ section.
...
+    If the directory, nor the original configuration directory (see below) do
+    not exist, mpv tries to create and use it.
+
+``~/.mpv/``
+    The original (pre 0.5.0) configuration directory. It will continue to be
+    read if present.
+
+    If both this directory and the standard configuration directory are
+    present, configuration will be read from both with the standard
+    configuration directory content taking precedence.
```
Comment (sfan5) [inline DOCS/man/mpv.rst:1526]: Since 22a27c6720244dd24c9b704c8a8225ca1665272f there's a warning if this is the case, might be worth noting here

**Comment by maddovr** [comment]
While we are at it, can we get proper XDG support by setting `icc-cache-dir` and `gpu-shader-cache-dir` to `~/.cache/mpv` and `watcher-later-directory` to `~/.local/share/mpv` (as suggested by @haasn in [this](https://github.com/mpv-player/mpv/commit/269f0e743e5634691f0c9d5b1b8a4bb68eedbbd0#commitcomment-40509344) comment)?

**Comment by Dudemanguy** [comment]
Probably out of scope for this PR but using the cache directory makes sense.

**Comment by philipl** [comment]
It's reasonable to want to use the cache dir. It's not reasonable to want to do that in this PR.

**Review by jeeb — APPROVED** [review]
LGTM

**Pull request merged**